### PR TITLE
mgr/dashboard: upgraded python dev dependencies

### PR DESCRIPTION
**Pull request by alfonsomthd**
* pylint, astroid: upgraded only for python 3.
  Latest stable, compatible even with python 3.7

* pycodestyle: 2.4.0 latest stable (python 2 & 3)

* Fixed linting/style issues found after upgrade.

Fixes: https://tracker.ceph.com/issues/37385

Signed-off-by: Alfonso Martínez <almartin@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by LenzGr** [comment]
Thank you for taking care of this, @alfonsomthd! Would you mind splitting this PR into two separate commits? I think it would make sense to have one that updates the versions and configuration of our environment, and then a separate cleanup commit that addresses all of the things that have been reported.

**Comment by alfonsomthd** [comment]
@LenzGr Sure. Done.

File: src/pybind/mgr/dashboard/tools.py [hunk 1]
```diff
@@ -832,6 +832,7 @@ def getargspec(func):
             func = func.__wrapped__
     except AttributeError:
         pass
+    # pylint: disable=deprecated-method
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/tools.py]: [How](https://github.com/pytest-dev/pluggy/pull/94/files) `pluggy` devs dealt with this...

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard/tools.py]: we're very likely going full Python 3 by the end of the year. I'd be ok with a pylint for the moment.

**Comment by callithea** [comment]
QA run was successful: http://pulpito.ceph.com/laura-2019-01-21_12:20:46-rados:mgr-wip-lpaduano-testing-26007-distro-basic-smithi/

**Pull request merged**